lib/ResultAggregator.ts

```diff
@@ -119,13 +130,54 @@ export class ResultAggregator implements IResultAggregator {
         nObsTimestamp[j]++;
       }
     }
+
+    const timestampsAverage = timestampsSum.map((ts, i) => ts / nObsTimestamp[i]);
+
+    for (const timestamps of timestampsAll) {
+      for (const [ j, ts ] of timestamps.entries()) {
+        timestampsStd[j] += (ts - timestampsAverage[j]) ** 2;
+      }
+    }
+
+    for (let i = 0; i < timestampsStd.length; i++) {
+      timestampsStd[i] = Math.sqrt(timestampsStd[i] / nObsTimestamp[i]);
+    }
+
     return {
       timestampsMax,
       timestampsMin,
-      timestampsAverage: timestampsSum.map((ts, i) => ts / nObsTimestamp[i]),
+      timestampsAverage,
+      timestampsStd,
     };
   }
 
+  public aggregateRawGroupedResults(
+    groupedResults: Record<string, IResult[]>,
+    aggregateResults: IAggregateResult[],
+  ): IRawAggregateResult[] {
+    const rawAggregateResults: IRawAggregateResult[] = [];
+    const aggregateResultsMap: Map<string, IAggregateResult> = new Map(
+      aggregateResults.map(result => [ `${result.name}:${result.id}`, result ]),
+    );
+
+    for (const [ id, resultsSet ] of Object.entries(groupedResults)) {
+      // There will be always an aggregate results because it has been made from the group results
+      const currentAggregateResults = aggregateResultsMap.get(id)!;
+      const currentRawAggregateResult: IRawAggregateResult = {
+        ...currentAggregateResults,
+        timeAggregate: [],
+      };
+      if (currentAggregateResults.error) {
+        currentRawAggregateResult.error = currentAggregateResults.error;
+      }
+      for (const { time, error } of resultsSet) {
+        currentRawAggregateResult.timeAggregate.push(error ? Number.NaN : time);
```

Comment:
So the `timeAggregate` field will contain the individual `time` fields for a given query, from which the `timeMin` and `timeMax` are taken, as well as the `time` average?

Reply:
> So the `timeAggregate` field will contain the individual `time` fields for a given query, from which the `timeMin` and `timeMax` are taken, as well as the `time` average?

I am not sure I understand, I think it simply takes all the `time`, so `timeMax` and `timeMin` will be part of it but `time` average maybe not depending on the distribution since it is a computed value.